betty/extension/_webpack/__init__.py

```diff
@@ -0,0 +1,291 @@
+"""
+Integrate Betty with `Webpack <https://webpack.js.org/>`_.
+
+This module is internal.
+"""
+
+from __future__ import annotations
+
+from asyncio import to_thread
+from collections.abc import Callable, Sequence
+from hashlib import md5
+from json import dumps, loads
+from logging import getLogger
+from os.path import join
+from pathlib import Path
+from shutil import copytree, copy2
+from typing import Any
+
+import aiofiles
+from aiofiles.os import makedirs
+
+from betty._npm import NpmRequirement
+from betty.app.extension import Extension, UserFacingExtension
+from betty.asyncio import gather
+from betty.cache.file import BinaryFileCache
+from betty._npm import npm
+from betty.extension._webpack.jinja2.filter import FILTERS
+from betty.fs import iterfiles, ROOT_DIRECTORY_PATH
+from betty.generate import Generator, GenerationContext
+from betty.html import CssProvider
+from betty.jinja2 import Jinja2Provider
+from betty.job import Context
+from betty.locale import Str
+from betty.requirement import Requirement, AllRequirements, AnyRequirement
+
+# @todo Finish this
+# @todo - Determine where assets go now, preferably now scattered around and deeply nested in the source code
+# @todo - Update that path everywhere (.gitignore, setuptools config, etc?)
+PREBUILT_ASSETS_DIRECTORY_PATH = ROOT_DIRECTORY_PATH / "prebuild" / "webpack"
+
+
+class WebpackEntrypointProvider:
+    @classmethod
+    def webpack_entrypoint_directory_path(cls) -> Path:
```

Comment:
What if we move the files to the assets, but for each extension use its name in the path? Then we can remove this method entirely.